vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/AbstractGridMultiSelectionModel.java

```diff
@@ -113,9 +113,18 @@ public void selectFromClient(T item) {
             fireSelectionEvent(new MultiSelectionEvent<>(getGrid(),
                     getGrid().asMultiSelect(), oldSelection, true));
 
-            long size = getDataProviderSize();
+            if (!isSelectAllCheckboxVisible()) {
+                // Skip changing the state of Select All checkbox if it was
```

Comment:
I only noticed now that this condition is absent in all the other methods concerning the selection. Why? If it is an optimization, I would expect it to be in all the methods. This question also comes to the original PR for V23.

Reply:
I though about this as well, but it was like that before and the intention of the PR is to add the indeterminate state, not to optimize performance, so I left as is.